runtime/common/src/main/kotlin/kotlinx/serialization/modules/SerialModuleCollector.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+@file:Suppress("RedundantVisibilityModifier")
+
+package kotlinx.serialization.modules
+
+import kotlinx.serialization.*
+import kotlin.reflect.KClass
+
+/**
+ * Collector can be used to inspect or accumulate content of some [SerialModule],
+ * using a visitor-like pattern: [contextual] and [polymorphic] functions
+ * are invoked for each registered serializer.
+ *
+ * @see [SerialModule.dumpTo].
+ */
+public interface SerialModuleCollector {
+
+    /**
+     * Passes to this collector a serializer, associated with [kClass] for contextual
+     * serialization.
+     *
+     * @see ContextSerializer
+     */
+    public fun <T : Any> contextual(kClass: KClass<T>, serializer: KSerializer<T>)
+
+    /**
+     * Passes to this collector a serializer for [actualClass], registered for polymorphic serialization
+     * in the scope of [baseClass].
+     *
+     * @see PolymorphicSerializer
+     */
+    public fun <Base : Any, Sub : Base> polymorphic(
+        baseClass: KClass<Base>,
+        actualClass: KClass<Sub>,
+        actualSerializer: KSerializer<Sub>
+    )
+}
+
+/**
+ * Version of [SerialModuleCollector.contextual] with reified argument.
+ */
+public inline fun <reified T: Any> SerialModuleCollector.contextual(serializer: KSerializer<T>) {
```

Comment:
Members are usually more discoverable, it is better to add it as a member (and for polymorphic)

Reply:
Why so? I think there is no difference for auto-completion when you start typing `contex...`. Also, adding this method to `SerialModuleCollector` interface would make it overridable (even with default impl), which I believe is not desirable.

PolymorphicBuilder already has reified version of `addSubclass`.

Reply:
>PolymorphicBuilder already has reified version of addSubclass.

I mean `polymorphic` function itself.

I agree this should not be a member